packages/ckeditor5-engine/src/model/utils/insertcontent.ts

```diff
@@ -88,7 +116,35 @@ export default function insertContent(
 
 		insertion.handleNodes( nodesToInsert );
 
-		const newRange = insertion.getSelectionRange();
+		let newRange = insertion.getSelectionRange();
+
+		if ( fakeMarkerElements.length ) {
+			const markerRange = newRange && LiveRange.fromRange( newRange );
```

Comment:
For readability reasons, I'd prefer `= newRange ? LiveRange.fromRange( newRange ) : null`.